toolkit/tools/scheduler/scheduler.go

```diff
@@ -277,6 +277,26 @@ func startWorkerPool(agent buildagents.BuildAgent, workers, buildAttempts, check
 	return
 }
 
+// debugStuckNode is a debugging function that will print out the stuck node and all nodes that are blocking it.
+func debugStuckNode(buildState *schedulerutils.GraphBuildState, pkgGraph *pkggraph.PkgGraph, stuckNode *pkggraph.PkgNode, indent int) {
+	isStuck := !buildState.IsNodeAvailable(stuckNode)
```

Comment:
Since these chains may be long/deep, I'd focus only on the stuck nodes. I think we don't care much about the ones, which were fine and we can make reading this output simpler on the developer:
```suggestion
	if buildState.IsNodeAvailable(stuckNode) {
		return
	}
```